Source/WebDriver/glib/SessionHostGlib.cpp

```diff
@@ -134,43 +134,58 @@ static guint16 freePort()
 
 void SessionHost::launchBrowser(Function<void (std::optional<String> error)>&& completionHandler)
 {
+    String targetIp;
+    uint16_t targetPort = 0;
+
+    if (!m_targetIp.isEmpty() && m_targetPort) {
+        targetIp = m_targetIp;
+        targetPort = m_targetPort;
+    } else if (m_capabilities.targetAddr && m_capabilities.targetPort) {
+        targetIp = m_capabilities.targetAddr.value();
+        targetPort = m_capabilities.targetPort.value();
+    }
+
     m_cancellable = adoptGRef(g_cancellable_new());
-    GRefPtr<GSubprocessLauncher> launcher = adoptGRef(g_subprocess_launcher_new(G_SUBPROCESS_FLAGS_NONE));
-    guint16 port = freePort();
-    GUniquePtr<char> inspectorAddress(g_strdup_printf("127.0.0.1:%u", port));
-    g_subprocess_launcher_setenv(launcher.get(), "WEBKIT_INSPECTOR_SERVER", inspectorAddress.get(), TRUE);
+    GUniquePtr<char> inspectorAddress(
+        g_strdup_printf("%s:%u", targetIp.isEmpty()?"127.0.0.1":targetIp.latin1().data(), targetPort > 0 ? targetPort :freePort())
+    );
+    if (targetIp.isEmpty()) {
```

Comment:
We can return early here instead and avoid all the re-indentation.

Reply:
There's connectToBrowser call at the end of the function that still needs to be called in both cases

Reply:
Fixed